compiler/rustc_mir_build/src/build/matches/test.rs

```diff
@@ -494,7 +494,14 @@ impl<'a, 'tcx> Builder<'a, 'tcx> {
         }
 
         let eq_def_id = self.tcx.require_lang_item(LangItem::PartialEq, Some(source_info.span));
-        let method = trait_method(self.tcx, eq_def_id, sym::eq, [ty, ty]);
+
+        let mut args: Vec<ty::GenericArg<'tcx>> = vec![ty.into(), ty.into()];
+        // If `PartialEq` is `#[const_trait]`, then add a const effect param
+        if self.tcx.generics_of(eq_def_id).host_effect_index.is_some() {
```

Comment:
Maybe I should make a helper fn that appends a const param and interns the generic args list, something like:

```
fn with_opt_const_effect_param(tcx: TyCtxt<'tcx>, caller_def_id: LocalDefId, callee_def_id: DefId, args: impl IntoIterator<Item = impl Into<GenericArg>>) -> &'tcx GenericArgs<'tcx> {}
```

That checks if the callee_def_id has a const effect param, and if so, constructs it based off of the caller_def_id, appends it, then interns the args...

Reply:
Yes, that would be very nice for more lang item migrations.

Reply:
Yeah, it ended up being a lot better in #118386. I'll uplift it into this PR, which I think we can land after #118282.